Link login modal to follow a post

dinaelhanan opened the pull request:
Fixes #4152

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

dinaelhanan commented:
@publiclab/reviewers

oorjitchowdhary commented:
@sidharthbansal aren't #4437 and #4152 the same issues? 

@dinaelhanan I think you have made changes to a wrong file..
I don't think this is the file for the buttons @sidharthbansal was talking about

oorjitchowdhary commented:
https://github.com/publiclab/plots2/blob/699108b54624cf0554c6bb87d4f05724f79901ad/app/views/like/_like.html.erb 
This is the file for the buttons

dinaelhanan commented:
@oorjitchowdhary thanks!

SidharthBansal commented:
@dinaelhanan please rebase and resolve conflicts so that @oorjitchowdhary or I or anyone else can help you further.
Also, please provide the gifs.
Thanks

dinaelhanan commented:
@SidharthBansal I am trying on my localhost and it's not working, I click the button and nothing happens..Please check out my changes. Thanks

dinaelhanan commented:
@publiclab/reviewers would appreciate some guidance on this, thanks!

SidharthBansal commented:
My laptop crashed yesterday again. Now it is fine. Sorry was not able to
help you. I will help you after getting up from sleep.

On Fri, Jan 4, 2019, 8:12 AM d1g1t4ld1n4 <notifications@github.com wrote:

> @publiclab/reviewers <https://github.com/orgs/publiclab/teams/reviewers>
> would appreciate some guidance on this, thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4464#issuecomment-451340015>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUACQ6xjHfdjvNHTCLj2w-EFf2PL5WZqks5u_r-LgaJpZM4Zk-ox>
> .
>

oorjitchowdhary commented:
I think we should wait on this until #4453 is merged. That may be the issue here..

SidharthBansal commented:
Yeah we need to merge that before this PR.
Thanks Oorjit. Dina you can work on Amy other issue in the meantime.

On Fri, Jan 4, 2019, 11:38 AM Oorjit Chowdhary <notifications@github.com
wrote:

> I think we should wait on this until #4453
> <https://github.com/publiclab/plots2/pull/4453> is merged. That may be
> the issue here..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4464#issuecomment-451360585>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AUACQww41rTRfA2tG5DeuflP6BtBdMikks5u_u_ygaJpZM4Zk-ox>
> .
>

oorjitchowdhary commented:
#4453 is merged.. @dinaelhanan update your branch.. And try adding the 'requireLogin' class to the follow button.. It should work then

dinaelhanan commented:
@oorjitchowdhary OK I've updated and added requireLogin class to the buttons. Please checkout my changes, I'm not quite sure if I've added the class correctly. Thanks for all the help :+1:

oorjitchowdhary commented on app/views/like/_like.html.erb:
```diff
@@ -10,6 +10,19 @@
+       <% end %>
+  <li rel="tooltip" title="Follow" class="btn btn-default btn-sm" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if c ...
+      <i class="fa fa-user-plus" aria-hidden="true"></i> 
     <li title="Follow" class="btn btn-default btn-sm" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user  ...
```
```diff
- <li title="Follow" class="btn btn-default btn-sm" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user && !current_user.following?(node.author) %>&nbsp;<a class='btn btn-sm' href='/relationships?followed_id=<%= node.author.id %>' data-method='post' > <%= node.author.name %></a><% else %><%= node.author.name %><% end %>" title="Follow by tag or author" data-placement="left">
+ <li title="Follow" class="btn btn-default btn-sm requireLogin" data-html="true" rel="popover" data-content="<%= "No tags" if tagnames.nil? || tagnames.length == 0 %><% if tagnames %><% tagnames.each do |tagname| %><p style='margin-bottom:3px;'><a href='/subscribe/tag/<%= tagname %>' class='btn btn-default btn-mini'><%= tagname %></a></p><% end %><% end %><hr /><i class='fa fa-user'></i><% if current_user && !current_user.following?(node.author) %>&nbsp;<a class='btn btn-sm' href='/relationships?followed_id=<%= node.author.id %>' data-method='post' > <%= node.author.name %></a><% else %><%= node.author.name %><% end %>" title="Follow by tag or author" data-placement="left">
```

oorjitchowdhary replied:
Remove all the added lines and just add `requireLogin` in the class as shown in the diff above and in that line only.. And then try it out

This is only a line change..

dinaelhanan replied:
@oorjitchowdhary Ok thanks for that, I've done it :+1:

jywarren commented:
In the rails console (`rails console`) you can run `User.find(1).unban` --
sorry, there's an open issue where we're trying to debug this! #608 i
think? Bug recently cropped up!

On Sat, Jan 5, 2019 at 7:22 PM d1g1t4ld1n4 <notifications@github.com> wrote:

> @oorjitchowdhary <https://github.com/oorjitchowdhary> ok I think I've
> done the correct changes. I would add a gif showing the changes but when
> testing the spam button, I accidently got the admin account banned on my
> local version, how would I fix this? Thanks.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4464#issuecomment-451704215>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ7nFQIQXl-09d-vA6QYzdT9EwGDZks5vAUGmgaJpZM4Zk-ox>
> .
>

dinaelhanan commented:
@jywarren ok thanks for that! I've done it and it works. so this is what I'm getting. I think that's what I was supposed to do:

![followflow](https://user-images.githubusercontent.com/19677511/50739108-5da8ce80-11a9-11e9-8213-f1eea7bddc46.gif)

dinaelhanan commented:
@SidharthBansal I did click on the follow button, that little "blog" button dialog pops up after you click follow on the local version and the actual site (this gif is from the actual site. On the previous gif, you can see I did the same and clicked on what part of the note i would like to follow):
![screencast20190106140438](https://user-images.githubusercontent.com/19677511/50740432-50e1a600-11bc-11e9-9654-bcb0a01f8ead.gif)

SidharthBansal commented:
Checking this on my laptop.

SidharthBansal commented:
Pushed these changes on the unstable branch

SidharthBansal commented:
It is working.
Thanks Dina. Merging it up @jywarren.
